gate-web/src/main/groovy/com/netflix/spinnaker/gate/config/SlackConfig.groovy

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.gate.config
+
+import com.netflix.spinnaker.gate.services.SlackService
+import groovy.transform.CompileStatic
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import retrofit.Endpoint
+import retrofit.RequestInterceptor
+import retrofit.RestAdapter
+import retrofit.client.OkClient
+import retrofit.converter.JacksonConverter
+
+import static retrofit.Endpoints.newFixedEndpoint
+
+@Configuration
+@CompileStatic
+@ConditionalOnProperty('slack.token')
+class SlackConfig {
+
+  @Value('${slack.token}')
+  String token
+
+  @Bean
+  Endpoint slackEndpoint(
+    @Value('${slack.base-url}') String slackBaseUrl) {
+    newFixedEndpoint(slackBaseUrl)
+  }
+
```

Comment:
RequestInterceptor not used anywhere else , may be it can be moved into slackService definition itself. 

Looks like we do this in every controller. n/m.

Reply:
> Few minor observations. Also tests are missing :).

Thanks for the review! Tests are coming, I wanted some initial feedback first :)